cpmpy/solvers/pumpkin.py

```diff
@@ -0,0 +1,550 @@
+
+            if cpm_expr.name == "==": comp = Comparator.Equal
+            if cpm_expr.name == "<=": comp = Comparator.LessThanOrEqual
+            if cpm_expr.name == ">=": comp = Comparator.GreaterThanOrEqual
+            if cpm_expr.name == "!=": comp = Comparator.NotEqual
+            if cpm_expr.name == "<":  comp, rhs = Comparator.LessThanOrEqual, rhs - 1
+            if cpm_expr.name == ">":  comp, rhs = Comparator.GreaterThanOrEqual, rhs + 1
+        else:
+            raise ValueError(f"Cannot convert CPMpy expression {cpm_expr} to a predicate")
+
+        var = self.to_pum_ivar(lhs)
+        return Predicate(var, comp, rhs)
+
+
+    def to_pum_ivar(self, cpm_var):
+        """
+            Helper function to convert (boolean) variables and constants to Pumpkin integer expressions
+        """
+        if is_any_list(cpm_var):
+            return [self.to_pum_ivar(v) for v in cpm_var]
+        elif isinstance(cpm_var, _BoolVarImpl):
+            return self.pum_solver.boolean_as_integer(self.solver_var(cpm_var))
+        elif is_num(cpm_var):
+            return self.solver_var(intvar(cpm_var, cpm_var))
+        else:
+            return self.solver_var(cpm_var)
+
+
+    def _sum_args(self, expr, negate=False):
+        """
+            Helper function to convert CPMpy sum-like operators into pumpkin-compatible arguments.
+            expr is expected to be a `sum`, `wsum` or `sub` operator.
+
+            :return: Returns a list of Pumpkin integer expressions
+        """
+        args = []
+        if isinstance(expr, Operator) and expr.name == "sum":
+            for cpm_var in expr.args:
+                pum_var = self.solver_var(cpm_var)
+                if cpm_var.is_bool(): # have convert to integer
+                    pum_var = self.pum_solver.boolean_as_integer(pum_var)
+                args.append(pum_var.scaled(-1 if negate else 1))
+        elif isinstance(expr, Operator) and expr.name == "wsum":
+            for w, cpm_var in zip(*expr.args):
+                if w == 0: continue # exclude
+                pum_var = self.solver_var(cpm_var)
+                if cpm_var.is_bool(): # have convert to integer
+                    pum_var = self.pum_solver.boolean_as_integer(pum_var)
+                args.append(pum_var.scaled(-w if negate else w))
+        elif isinstance(Operator, expr) and expr.name == "sub":
+            x, y = self.solver_vars(expr.args)
+            if expr.args[0].is_bool():
+                x = self.pum_solver.boolean_as_integer(x)
+            if expr.args[1].is_bool():
+                y = self.pum_solver.boolean_as_integer(y)
+            args = [x.scaled(-1 if negate else 1), y.scaled(1 if negate else -1)]
+        else:
+            raise ValueError(f"Unknown expression to convert in sum-arguments: {expr}")
+        return args
+
+    def _get_constraint(self, cpm_expr):
+        """
+            Get a solver's constraint by a supported CPMpy constraint
```

Comment:
'by a'? confusing

Reply:
also, add to the docs "expects a transformed cpm_expr as input, this is a separate function so we can add an indicator variable to it if needed"